jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/UpnpDIDLFactory.java

```diff
@@ -183,6 +198,63 @@ private MimeType getMimeType(MediaFile song, Player player) {
         return mimeTypeString == null ? null : MimeType.valueOf(mimeTypeString);
     }
 
+    private StorageFolder createMusicFolder(int id, String name, ProcId procId, int childCount) {
+        StorageFolder container = new StorageFolder();
+        container.setId(procId.getValue() + ProcId.CID_SEPA + id);
+        container.setParentID(procId.getValue());
+        container.setTitle(name);
+        container.setChildCount(childCount);
+        return container;
+    }
+
+    public StorageFolder toMusicFolder(MusicFolder folder, ProcId procId, int childCount) {
+        return createMusicFolder(folder.getId(), folder.getName(), procId, childCount);
+    }
+
+    public StorageFolder toMusicFolder(MediaFile folder, ProcId procId, int childCount) {
+        return createMusicFolder(folder.getId(), folder.getName(), procId, childCount);
+    }
+
+    public GenreContainer toMusicIndex(MusicIndex musicIndex, ProcId procId, int childCount) {
+        GenreContainer container = new GenreContainer();
+        container.setId(procId.getValue() + ProcId.CID_SEPA + musicIndex.getIndex());
+        container.setParentID(procId.getValue());
+        container.setTitle(musicIndex.getIndex());
+        container.setChildCount(childCount);
+        return container;
+    }
+
+    public GenreContainer toGenre(Genre genre, ProcId procId, boolean isCountVisible, int childCount) {
+        GenreContainer container = new GenreContainer();
+        container.setId(procId.getValue() + ProcId.CID_SEPA + genre.getName());
+        container.setParentID(procId.getValue());
+        container.setTitle(isCountVisible ? genre.getName().concat(" ").concat(Integer.toString(genre.getSongCount()))
+                : genre.getName());
+        container.setChildCount(childCount);
+        return container;
+    }
+
+    public PlaylistContainer toPlaylist(Playlist playlist) {
+        PlaylistContainer container = new PlaylistContainer();
+        container.setId(ProcId.PLAYLIST.getValue() + ProcId.CID_SEPA + playlist.getId());
+        container.setParentID(ProcId.PLAYLIST.getValue());
+        container.setTitle(playlist.getName());
+        container.setDescription(playlist.getComment());
+        container.setChildCount(playlist.getFileCount());
+        container.addProperty(toPlaylistArt(playlist));
+        return container;
+    }
+
+    public MusicArtist toArtist(MediaFile artist, int childCount) {
+        MusicArtist container = new MusicArtist();
+        container.setTitle(artist.getName());
+        container.setId(ProcId.FOLDER.getValue() + ProcId.CID_SEPA + artist.getId());
+        mediaFileService.getParent(artist).ifPresent(parent -> container.setParentID(String.valueOf(parent.getId())));
+        artist.getCoverArtPath().ifPresent(path -> container.addProperty(toArtistArt(artist)));
```

Comment:
All methods that generate cover art can be changed to private.

 - toArtistArt
 - toAlbumArt
 - toPodcastArt
 - toPlaylistArt

Reply:
Done.